include/range/v3/utility/iterator.hpp

```diff
@@ -619,6 +619,64 @@ namespace ranges
             Char const *delim_;
         };
 
+        template <typename Delim, typename Char = char,
+                  typename Traits = std::char_traits<Char>>
+        struct ostream_joiner
+        {
+            using difference_type = std::ptrdiff_t;
+            using char_type       = Char;
+            using traits_type     = Traits;
+            using ostream_type    = std::basic_ostream<Char, Traits>;
+
+            constexpr ostream_joiner() = default;
+            ostream_joiner(ostream_type &s, Delim const &d) noexcept
+              : delim_(d), sout_(std::addressof(s)), first_(true)
```

Comment:
The `noexcept` is curious here, since `Delim` is not constrained to be a nothrow-copyable type. Is that really what is being standardized? My preference is to nuke the `noexcept` on the two constructors and on `make_ostream_joiner_fn::operator()`.

Reply:
You are right. I removed it

Reply:
I also had forgotten the noexcept on `operator++` and `operator*`...